### ISSUE TITLE
TCP behaivor

### ISSUE THREAD

**Issue by Spikeophant** [coturn/coturn#351]
Hello, I cannot seem to get tcp handoff to work right.  I can get it to work via UDP, however our webrtc machines are directly exposed to the internet so I just need this to work to do tcp tunneling if UDP is blocked on the remote side.  Is the best way to do this to force the system into tcp mode?  Or am I missing something in my configs?  Thanks!

**Comment by misi** [comment]
Can you explain more detailed the issue. 
I didn't fully understand it yet.

**Comment by Spikeophant** [comment]
Okay, no problem.

I have a coturn server setup and exposed to the internet.  I have defined tcp and udp ports for it to listen on, and can validate it is listening.  However when I block UDP inbound/outbound (excepting DNS traffic) from a test machine, I cannot get it to connect to the coturn server in TCP to hand off the traffic and make a webrtc call.

**Comment by misi** [comment]
What do you see in your browsers webrtc trace?

**Comment by benbro** [comment]
I'm also not getting TCP candidates after upgrading from Ubuntu 18.04 (coturn 4.5.0.7) to Ubuntu 19.04 (coturn 4.5.1.0).

This gives me udp candidates: turn:myserver.com:3478?transport=udp
This doesn't give me tcp candidates: turn:myserver.com:3478?transport=tcp
This also doesn't give me tcp candidates: turns:myserver.com:443?transport=tcp

Did something in turnserver.conf changed?
Is there something in the log that says if udp/tcp listeners are active?

**Comment by Spikeophant** [comment]
My issue turned out to be some funky nat pinning we needed to do.  I am good to go, sorry I should have closed this earlier.  @benbro  you should open a new ticket, but you might look at the output of netstat and see if the box is listening on the desired ports.
